>From 941072cd481b4c070b1a5cc67dd1ad7ab0ebf880 Mon Sep 17 00:00:00 2001
From: Jochen Keil <jochen.keil@emlix.com>
Date: Fri, 29 Jan 2010 10:30:08 +0100
Subject: [PATCH 1/9] require at least one white space and/or tab after a keyword so we won't
 keywords in function names e.g. proc foo_set_var {}

---
 tcl-dox.l |   17 ++++++++---------
 1 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/tcl-dox.l b/tcl-dox.l
index 2dd813f..6d1a1ff 100644
--- a/tcl-dox.l
+++ b/tcl-dox.l
@@ -83,22 +83,21 @@ WS    [ \t]
  }
 
 
-
- "class" {
+ "class"[ \t]+ {
     ECHO;
     BEGIN(CLASS);
  } 
 
- "constructor" {
+ "constructor"[ \t]+ {
     fprintf(yyout, "public:\n %s", g_className);
     BEGIN(PROC_OPEN);
  }
 
- "destructor" {
+ "destructor"[ \t]+ {
     fprintf(yyout, "public:\n ~%s()", g_className);
  }
 
- "proc" {
+ "proc"[ \t]+ {
     /* If we are inside of a class then the proc is static. */
     if(g_className)
     {
@@ -111,22 +110,22 @@ WS    [ \t]
     BEGIN(PROC);
  }
 
- "method" {
+ "method"[ \t]+ {
     fprintf(yyout, "%s", "string ");
     BEGIN(PROC);
  }
 
- "namespace eval" {
+ "namespace eval"[ \t]+ {
     fprintf(yyout, "%s", "namespace");
     BEGIN(NAMESPACE);
  }
 
- "public"|"private"|"protected" {
+ "public"|"private"|"protected"[ \t]+ {
     ECHO;
     fprintf(yyout, "%s", ":\n");
  }
 
- "variable"|"common"|"global"|"array set"|"set" {
+ "variable"|"common"|"global"|"array set"|"set"[ \t]+ {
     fprintf(yyout, "%s", "type ");
     BEGIN(VAR);
  }
-- 
1.6.6.1

